File: scratch/run_git.py
import subprocess
import os
import sys

cwd = r"d:\OneDrive\2025 - 2026\Hoc tap Lop 5"
out_file = r"d:\OneDrive\2025 - 2026\Hoc tap Lop 5\scratch\git_output.txt"

env = os.environ.copy()
env["GIT_TERMINAL_PROMPT"] = "0"
env["GCM_INTERACTIVE"] = "never"
env["GIT_OPTIONAL_LOCKS"] = "0"

commands = [
    ["git", "status"],
    ["git", "remote", "-v"],
    ["git", "branch", "-a"]
]

with open(out_file, "w", encoding="utf-8") as f:
    f.write("STARTING...\n")

res = []
for cmd in commands:
    res.append(f"=== Running: {' '.join(cmd)} ===")
    try:
        p = subprocess.run(cmd, cwd=cwd, capture_output=True, text=True, timeout=3, env=env)
        res.append("STDOUT:\n" + p.stdout)
        res.append("STDERR:\n" + p.stderr)
    except Exception as e:
        res.append(f"ERROR: {e}")

with open(out_file, "w", encoding="utf-8") as f:
    f.write("\n".join(res))

print("FINISHED PYTHON SCRIPT", flush=True)

File: scratch/run_git.js
const { execSync } = require('child_process');
const fs = require('fs');
const path = require('path');

const outFile = path.join(__dirname, 'git_output.txt');
let log = [];

function run(cmd) {
    log.push(`=== Running: ${cmd} ===`);
    try {
        const out = execSync(cmd, {
            cwd: path.join(__dirname, '..'),
            encoding: 'utf-8',
            timeout: 5000,
            env: { ...process.env, GIT_TERMINAL_PROMPT: '0', GCM_INTERACTIVE: 'never' }
        });
        log.push("STDOUT:\n" + out);
    } catch (e) {
        log.push("ERROR:\n" + (e.stdout || '') + "\n" + (e.stderr || '') + "\n" + e.message);
    }
}

run('git status');
run('git remote -v');
run('git log -n 5 --oneline');

fs.writeFileSync(outFile, log.join('\n\n'), 'utf-8');
console.log("DONE NODE SCRIPT");
